receiver/splunkhecreceiver/factory.go

```diff
@@ -68,6 +68,7 @@ func createMetricsReceiver(
 	if err != nil {
 		return nil, err
 	}
+	r.Component.(*splunkReceiver).metricsConsumer = consumer
```

Comment:
Just for readability 
```suggestion
	r.Unwrap().(*splunkReceiver).metricsConsumer = consumer
```

Reply:
It cannot be unexported, it needs to inherit the `Component` interface implementation